src/quick/items/qquickitemview.cpp

```diff
@@ -1512,7 +1512,7 @@ QQuickItemViewPrivate::~QQuickItemViewPrivate()
 
 bool QQuickItemViewPrivate::isValid() const
 {
-    return model && model->count() && model->isValid();
+    return model && ((model->count() && model->isValid()) || !visibleItems.isEmpty());
```

Comment:
This worries me. We use isValid() in many places, some of which will surely assume a valid model if this returns true.